docs/man/man3/hm2_pktuart_read.3hm2

```diff
@@ -0,0 +1,69 @@
+\# Author Boris Skegin
+\# Issued under the terms of the GPL v2 License or any later version
+.TH hm2_pktuart_read "3hm2" "2016-02-26" "LinuxCNC Documentation" "Hostmot2"
+.SH NAME
+
+hm2_pktuart_read \- read data from a Hostmot2 UART buffer
+
+.SH SYNTAX
+.HP
+int hm2_pktuart_read(char *name,  unsigned char data[], rtapi_u8 *num_frames, rtapi_u16 *max_frame_length, rtapi_u16 frame_sizes[])
+
+.SH DESCRIPTION
+\fBhm2_pktuart_read\fR reads data from the PktUART "name".
+"name" is a unique string given to each PktUART during hostmot2 setup. The names of 
+the available channels are printed to standard output during the driver loading 
+process and take the form:
+hm2_<board name>.<board index>.pktuart.<index> For example hm2_5i25.0.pktuart.0
+
+This function reads a variable number of PktUART packets from the the specified 
+channel. It should be used inside a realtime HAL component registered with the 
+main hostmot2 driver using the function hm2_pktuart_setup in the setup
+code. 
+
+"(*num_frames)*(*max_frame_length)" should be <= sizeof "data", which one tries
+to estimate or guess before calling the function.
+If there are more bytes in the buffer than the size of data array is, then
+RxArraySizeError is returned.
+
+Note that the PktUART MaxFrameSize is 1024 bytes as hard-coded in hostmot2.vhd .
+
+
+.SH RETURN VALUE
+Returns the number of bytes read on success and negative error codes on failure.
+
+"num_frames" which pointer is passed by value is set to the number of successfully 
+datagrams read.
+
+Negative error codes are:
+.TP 
+.B -1 - low level read/write error
+.TP
+.B -EINVAL - any PktUART configuration error per instance
```

Comment:
I do understand your argument, but I could find  any example macro for EINVAL
except  `#define EINVAL          22 `.  

Actually such macros are there, namely in: https://github.com/sirop/machinekit/blob/PktUART_LinuxCNC/src/hal/drivers/mesa-hostmot2/pktuart.c#L30-L45
```
// Tx mode register errors
#define  TxSCFIFOError  (214)  // Tx Send Count FIFO Error


// Rx mode register errors
#define RxRCFIFOError   (114)  // RCFIFO Error, Bit  4  
#define RxOverrunError  (111)  // Overrun error (no stop bit when expected) (sticky), Bit  1
#define RxStartbitError (110)  // False Start bit error (sticky), Bit  0

// Rx count register errors
#define RxPacketOverrrunError (1115)    // Bit 15         Overrun error in this packet
#define RxPacketStartbitError (1114)    // Bit 14         False Start bit error in this packet

// the next two error conditions
// are very unprobable, but we consider them nevertheless
#define RxPacketSizeZero      (1120)    // the length of the received packet is 0
#define RxArraySizeError      (1140)    // sizeof(data array)= num_frames*max_frame_length is too small for all the data in the buffer
```
Is this man file so confusing about that?
Shall I move these `define` macros to a header file?